soroban-sdk-macros/src/derive_client.rs

```diff
@@ -170,9 +170,18 @@ pub fn derive_client_impl(crate_path: &Path, name: &str, fns: &[syn_ext::Fn]) ->
                 .map(|t| {
                     let ident = match syn_ext::fn_arg_ident(t) {
                         Ok(ident) => ident,
-                        Err(_) => {
-                            errors.push(Error::new(t.span(), "argument not supported"));
-                            format_ident!("")
+                        Err(e) => {
+                            errors.push(e);
```

Comment:
I think we should make the error message generated by `fn_arg_ident` a little clearer so someone knows what is allowed, by changing the error from `argument not supported` to `argument in this form is not supported, use simple named arguments only`.

Reply:
Done